packages/cli/src/cli/cogify/action.make.cog.ts

```diff
@@ -0,0 +1,158 @@
+      required: false,
+    });
+    this.output = this.defineStringParameter({
+      argumentName: 'OUTPUT',
+      parameterShortName: '-o',
+      parameterLongName: '--output',
+      description: 'Output the job.json paths',
+      required: false,
+    });
+  }
+
+  async onExecute(): Promise<void> {
+    const logger = LogConfig.get();
+    const imagery = this.imagery.value;
+    const name = this.name.value;
+    if (imagery == null) throw new Error('Please provide a valid imagery source');
+    await Promise.all([3791, 3790, 3789, 3788].map((code) => ProjectionLoader.load(code)));
+
+    const source = imagery.endsWith('/') ? imagery : imagery + '/';
+    logger.info({ imagery: source }, 'FindImagery');
+    const imagerySetName = source.split('/').filter(Boolean).pop();
+
+    if (imagerySetName == null) throw new Error('Failed to find imagery set name');
+    const uri = await this.findImagery(source, logger);
+    const target = this.target.value;
+    const targets: string[] = target ? [target] : ['NZTM2000Quad', 'WebMercatorQuad'];
+
+    const outputs: string[] = [];
+    for (const target of targets) {
+      const id = ulid.ulid();
+      const tileMatrix = TileMatrixSets.find(target);
+      if (tileMatrix == null) throw new Error(`Cannot find tile matrix: ${target}`);
+      logger.info({ id, tileMatrix: tileMatrix.identifier }, 'SetTileMatrix');
+      const job = await this.makeCog(id, name ? name : imagerySetName, tileMatrix, uri);
+      outputs.push(job.getJobPath('job.json'));
+      // Write the output job json
+      const output = this.output.value;
+      if (output) {
+        fsa.write(output, JSON.stringify(outputs));
+      }
+    }
+  }
+
+  async makeCog(id: string, imageryName: string, tileMatrix: TileMatrixSet, uri: string): Promise<CogStacJob> {
+    let resampling;
+    /** Process Gebco 2193 as one cog of full extent to avoid antimeridian problems */
+    if (tileMatrix.projection === Epsg.Nztm2000 && imageryName.includes('gebco')) {
+      resampling = {
+        warp: 'nearest', // GDAL doesn't like other warp settings when crossing antimeridian
+        overview: 'lanczos',
+      } as const;
+    }
+
+    if (imageryName.includes('geographx')) {
+      resampling = {
+        warp: 'bilinear',
+        overview: 'bilinear',
+      } as const;
+    }
+
+    // Prepare the cutline
+    let cutline: { href: string; blend: number } | undefined;
+    if (this.cutline.value && this.blend.value) cutline = { href: this.cutline.value, blend: this.blend.value };
+    else if (this.cutline.value) new Error('Please provide a blend for the cutline');
+    else cutline = getCutline(imageryName);
+    if (cutline == null) throw new Error(`Cannot found default cutline from imagery name: ${imageryName}`);
+
+    const ctx: JobCreationContext = {
+      imageryName,
+      override: { id, projection: guessProjection(imageryName), resampling },
+      outputLocation: { type: 's3' as const, path: 's3://linz-basemaps' },
```

Comment:
we shouldnt force this to output to this bucket.

Reply:
Can we leave this as parameter?